src/Common/ErrorCodes.cpp

```diff
@@ -578,12 +579,57 @@ namespace ErrorCodes
 
     std::string_view getName(ErrorCode error_code)
     {
-        if (error_code >= END)
+        if (error_code < 0 || error_code >= END)
             return std::string_view();
         return error_codes_names.names[error_code];
     }
 
     ErrorCode end() { return END + 1; }
+
+    void increment(ErrorCode error_code, bool remote, const std::string & message, const std::string & stacktrace)
+    {
+        if (error_code >= end())
+        {
+            /// For everything outside the range, use END.
+            /// (end() is the pointer pass the end, while END is the last value that has an element in values array).
+            error_code = end() - 1;
+        }
+
+        ValuePair inc_value{
+            !remote,    /* local */
+            remote,     /* remote */
+            0,          /* error_time_ms */
+            message,    /* message */
+            stacktrace, /* stacktrace */
+        };
+        values[error_code].increment(inc_value);
+    }
+
+    /// ValuePair
+    ValuePair & ValuePair::operator+=(const ValuePair & value)
+    {
+        local  += value.local;
+        remote += value.remote;
+        message = value.message;
+        stacktrace = value.stacktrace;
+
+        const auto now = std::chrono::system_clock::now();
+        error_time_ms = std::chrono::duration_cast<std::chrono::milliseconds>(now.time_since_epoch()).count();
+
+        return *this;
+    }
```

Comment:
This method looks a little bit confusing to me (e.g. because we ignore `error_time_ms` from `ValuePair`, and also do not actually increment `message` and `stacktrace`).

I think I would rather do `ValuePairHolder::increment(bool remote, const std::string & message, const std::string & stacktrace)` which changes `ValuePair` fields itself.